options.go

```diff
@@ -77,6 +84,19 @@ func checkOptions(rp *Ringpop) []error {
 
 // Runtime options
 
+// Clock is used to set the Clock mechanism. Passing nil will use the system
+// clock. Testing harnesses will typically replace this with a mocked clock.
+func Clock(c clock.Clock) Option {
+	return func(r *Ringpop) error {
+		if c == nil {
```

Comment:
Why would we want/allow them to pass `nil`? Shouldn't passing nil be an error?

I think the system clock be the default option and then this functional option becomes optional.